model_compression_toolkit/target_platform_capabilities/target_platform/op_quantization_config.py

```diff
@@ -14,58 +14,135 @@
 # ==============================================================================
 
 import copy
-from typing import List
+from typing import List, Dict, Union, Any
 
 from mct_quantizers import QuantizationMethod
+from model_compression_toolkit.logger import Logger
 
 
-class OpQuantizationConfig:
+def clone_and_edit_object_params(obj: Any, **kwargs: Dict) -> Any:
     """
-    OpQuantizationConfig is a class to configure the quantization parameters of an operator.
+    Clones the given object and edit some of its parameters.
+
+    Args:
+        obj: An object to clone.
+        **kwargs: Keyword arguments to edit in the cloned object.
+
+    Returns:
+        Edited copy of the given object.
     """
 
+    obj_copy = copy.deepcopy(obj)
+    for k, v in kwargs.items():
+        assert hasattr(obj_copy,
+                       k), f'Edit parameter is possible only for existing parameters in the given object, ' \
+                           f'but {k} is not a parameter of {obj_copy}.'
+        setattr(obj_copy, k, v)
+    return obj_copy
+
+
+class AttributeQuantizationConfig:
+    """
+    Hold the quantization configuration of a weight attribute of a layer.
+    """
     def __init__(self,
-                 activation_quantization_method: QuantizationMethod,
                  weights_quantization_method: QuantizationMethod,
-                 activation_n_bits: int,
                  weights_n_bits: int,
                  weights_per_channel_threshold: bool,
                  enable_weights_quantization: bool,
+                 lut_values_bitwidth: Union[int, None],  # If None - set 8 in hptq, o.w use it
+                 ):
+        """
+        Initializes an attribute quantization config.
+
+        Args:
+            weights_quantization_method (QuantizationMethod): Which method to use from QuantizationMethod for weights quantization.
```

Comment:
An attr is a weight, so maybe we can remove the "weights" str from the variables

Reply:
I want to keep this inline with the terminology that we have in the node quantization configs that we build from the op config. So I prefer to keep the word "weights"